pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -2506,9 +2506,36 @@ func (r *ClickHouseReader) ListErrors(ctx context.Context, queryParams *model.Li
 
 	var getErrorResponses []model.Error
 
-	query := fmt.Sprintf("SELECT any(exceptionType) as exceptionType, any(exceptionMessage) as exceptionMessage, count() AS exceptionCount, min(timestamp) as firstSeen, max(timestamp) as lastSeen, any(serviceName) as serviceName, groupID FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU GROUP BY groupID", r.TraceDB, r.errorTable)
+	query := "SELECT any(exceptionMessage) as exceptionMessage, count() AS exceptionCount, min(timestamp) as firstSeen, max(timestamp) as lastSeen, groupID"
+	if len(queryParams.ServiceName) != 0 {
+		query = query + ", serviceName"
+	} else {
+		query = query + ", any(serviceName) as serviceName"
+	}
+	if len(queryParams.ExceptionType) != 0 {
+		query = query + ", exceptionType"
+	} else {
+		query = query + ", any(exceptionType) as exceptionType"
+	}
+	query += fmt.Sprintf(" FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU", r.TraceDB, r.errorTable)
 	args := []interface{}{clickhouse.Named("timestampL", strconv.FormatInt(queryParams.Start.UnixNano(), 10)), clickhouse.Named("timestampU", strconv.FormatInt(queryParams.End.UnixNano(), 10))}
-	if len(queryParams.OrderParam) != 0 {
+
+	if len(queryParams.ServiceName) != 0 {
+		query = query + " AND serviceName = @serviceName"
```

Comment:
are we sending the exact `serviceName` from frontend? Otherwise it should be `contains`?

Reply:
Yes we are matching exact keyword instead of using contains as mentioned in description: https://github.com/SigNoz/signoz/pull/1919#issue-1511012363

Reply:
Took this call after a discussion with @palashgdev 
Can change it to contains by using `ilike` function of clickhouse but then we won't have exactly equal operator.

Reply:
we cannot expect the user to write the fullname of the service to let him filter. Let me check the UI

Reply:
> then we won't have exactly equal operator
I didn't get this. Is this restriction from the UI?

Reply:
It's a restriction of using contains. 
If a user searches for `payment` then result will contain all the services containing `payment` keyword. Eg: `JusPayment` `paymentRazor` `payment`
In that case the `payment` service might not be top 10 results even after sorting. So user might miss payment service all together. Also currently we don't support multiple operators on UI so can't provide both exactly equal operator and contains operator

Reply:
what's the problem in showing the results and as the results will contain `payment` in the serviceName, they will understand to write better keywords. I am mostly against exact searches because service names can be upto 30-40 chars and users usually will type something unique about the service to filter.

Reply:
> In that case the payment service might not be top 10 results even after sorting. So user might miss payment service all together.

we do have pagination for this right?

Reply:
Ok will update it to use `ilike` operator